### DISCUSSION TITLE
Hero bonuses

### DISCUSSION THREAD

**Discussion by vincent-grosbois** [ihhub/fheroes2#2776] [Ideas]
Hello
I'm opening this thread to talk about adding Hero bonuses to H2, like in Heroes 3:
https://heroes.thelazy.net/index.php/List_of_hero_specialties

I think this could be a cool addition, and not very hard to implement (no UI elements required), and it would add lots of "customization" of Heroes, that are pretty much all the same in the same class in H2
The most complicated thing would be to design bonuses for all heroes, but we can probably find nice ideas or just reuse the type of bonuses there are in H3

**Comment by Branikolog** [comment]
Well, an idea is good, but  H2 game doesn't allow to choose any hero on start... So in my opinion, adding complicated and diverse bonuses is a bit unfair, as player cannot choose the needed hero...
On the other hand, it would be a nice, if different heroes have some more "otherness", besides a portrait only...
That are my thoughts: these bonuses shouldn't add much difference to the hero, but at the same time they should create a special behaviour of the hero. It could probably be an particular starting army. Or, we can make certain heroes to gain primary skills in their own, individual manner: like rising the chance of gaining att/def/pwr/klg by x% more, than other heroes.

**Reply by vincent-grosbois** [reply]
>as player cannot choose the needed hero...
that's not very hard to allow this in the UI for scenario :)

**Reply by Branikolog** [reply]
Oh, I'm not that person, who likes choosing a hero.) Each time playing H3 I set random hero, as it's more fun to start every scenario with a new hero. Better than choosing a hero with the best skill like logistics etc each scenario.) 
So, making an option in HoMMII UI is not necessary for me. (My personal point)  :D
But I'd be glad, if heroes would have some minor differences, making a gameplay each time a little bit different from another scenario.
For start we should make up a ton of features for all heroes, before we'd try to implement new GUI for it. ;)

**Comment by Csambodin** [comment]
Hello everyone! H-3 MoP versions I like it very much, implies clever modifications, his development progresses very slowly unfortunately and a lot the mistake :( . There is an opportunity H2ben captain to create who fights for the hero's beings and gives bonuses? Many interesting modifications can be found here...   https://vk.com/topic-118013293_34759455

**Reply by Branikolog** [reply]
Hi, @Csambodin !
For the current stage of the project we're not working on any drastic modifications of original H2 mechanics. But I'm not telling you, we won't in future.
There could be found thousands of proposals and suggestions on adding or changing something in Heroes games in the internet. So there's not enough just pasting a link here. If you have any brilliant idea, you should make at least some sort of text presentation of it, so all other people can read it and join conversation, whether this option is worth to be added. You can also join our discord channel or our social groups to talk to other people watching our project and to the developer team as well. But again, all improvements over the original game mechanics will be deeply discussed only after 1.0 release of fheroes2 project.

**Reply by Csambodin** [reply]
Dear @Branikolog ! Thank you for the answer. Not big thought ... H4 hero in the hero ( Kommanche) system modified... attack , Defense, Hit points, Damage, Magic power skill, Speed. From 6 abilities 4 eligible ones, these a combination gives additional bonuses. Speed + attacks it = Can Shoot.
A marriage is engagement between a woman and a man with a ring. The spouse receives an experience dot obtained in a fight.
On a map observation tower can be occupied, Kommanche protects it.
I wait for his 1.0 conclusions patiently. Perfect work !
